R/create_output_type_item.R

```diff
@@ -70,30 +82,39 @@ create_output_type_point <- function(output_type = c("mean", "median"),
     ))
   }
   output_type <- rlang::arg_match(output_type)
-  # Get output type id property according to config schema version
-  # TODO: remove back-compatibility with schema versions < v2.0.0 when support
-  # retired
-  config_tid <- hubUtils::get_config_tid(
-    config_version = hubUtils::get_schema_version_latest(schema_version, branch)
-  )
-
   schema <- download_tasks_schema(schema_version, branch)
+  pre_v4 <- hubUtils::version_lt("v4.0.0", schema_version = schema$`$id`)
 
   # create output_type_id
-  if (is_required) {
+  if (pre_v4 && is_required) {
     output_type_id <- list(output_type_id = list(
       required = NA_character_,
       optional = NULL
     ))
-  } else {
+  }
+  if (pre_v4 && !is_required) {
     output_type_id <- list(output_type_id = list(
       required = NULL,
       optional = NA_character_
     ))
   }
+  if (!pre_v4) {
+    output_type_id <- list(
+      output_type_id = list(
+        required = NA_character_
```

Comment:
I think initially this was used to ensure that when it was written out to JSON it would be written out as `["NA"]` so doesn't really have an impact on actual schema data types. But you're right! This will be obsolete shortly when this property changes to be `NULL`.  I mentioned in the PR description I wasn't tackling https://github.com/hubverse-org/hubAdmin/issues/72 in this PR but I probably should have just gone ahead and done that.